httpu/multiclient.go

```diff
@@ -68,3 +69,62 @@ func (mc *MultiClient) sendRequests(
 	}
 	return tasks.Wait()
 }
+
+// DoWithContext implements ClientInterfaceCtx.DoWithContext.
+//
+// If the underlying client(s) do not implement DoWithContext, an error is
+// returned before any requests are sent.
+func (mc *MultiClient) DoWithContext(
+	req *http.Request,
+	numSends int,
+) ([]*http.Response, error) {
+	var ctxDelegates []ClientInterfaceCtx
+	for _, del := range mc.delegates {
+		if cd, ok := del.(ClientInterfaceCtx); ok {
+			ctxDelegates = append(ctxDelegates, cd)
+		} else {
+			return nil, fmt.Errorf("client does not implement ClientInterfaceCtx")
+		}
+	}
```

Comment:
This is ugly because we have to check at runtime whether the `mc.delegates` members implement the `ClientInterfaceCtx` interface; if we put `DoWithContext` in the main `ClientInterface` interface, then we can avoid this.

Reply:
Is there a compelling case to have `ClientInterfaceCtx` include `ClientInterface`? If not, then we can have two separate `MultiClient` instances, one of which takes contexts.

Reply:
I think that `MultiClient` is only used (instantiated) by `goupnp.httpClient()`, which in turn is only used by `goupnp.DiscoverDevicesCtx()`, which ideally would pass its `ctx` into it.

So we could keep compatibility by having a context-respecting version of `MultiClient`, and also have `DiscoverDevicesCtx` use the context in one extra place.

Reply:
I started working on this, but `DiscoverDevicesCtx` calls `ssdp.SSDPRawSearchCtx` with the `httpClient`, so in order to get the Context behaviour, we'll either have to change the signature, or add another method that does the ~same thing but takes a `ClientInterfaceCtx`. Thoughts?

Reply:
Okay, pushed an approach that splits up the interface, and adds the aforementioned function. What do you think?

(also, separately: I've got an example using `NewWANIPConnection2ClientsCtx` that I've been using for testing since my router doesn't support `NewWANPPPConnection1Clients`; any interest in me adding that example to this PR, or as a follow-up?)

Reply:
Thanks, I'll try to take a look at this this evening (UK time).

As to your second question: that sounds like it should be a separate PR, but sounds good.